llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -12140,10 +12140,16 @@ SDValue DAGCombiner::foldVSelectOfConstants(SDNode *N) {
   for (unsigned i = 0; i != Elts; ++i) {
     SDValue N1Elt = N1.getOperand(i);
     SDValue N2Elt = N2.getOperand(i);
-    if (N1Elt.isUndef() || N2Elt.isUndef())
+    if (N1Elt.isUndef() || N2Elt.isUndef()) {
+      AllAddOne = false;
```

Comment:
This loop may have been trying to ignore undefs.

Reply:
Updated. This invalidates the assertion I added (since a vector of undefs can be both `AllAddOne` and `AllSubOne`) so I removed that as well.